### PR TITLE
Call master_exit_read_only in master initJob. Fix #57

### LINKED ISSUE #57: Support master_exit_read_only in Kubernetes operator
Since version 23.2 read_only flag in master's Hydra is persistent, so after update `master_exit_read_only` is required.

### PR DESCRIPTION
**Pull request by l0kix2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: pkg/components/master.go
Comment (zlobober) [inline pkg/components/master.go:162]: Wouldn't it break working with old images, if this command fails with "command maxter_exit_read_only not found"? Should we add "|| true" here?

**Review by zlobober — APPROVED** [review]

File: pkg/components/master.go
Comment (gritukan) [inline pkg/components/master.go:162]: Забавное наблюдение: этот вызов может пофейлиться не только из-за неподдержанного вызова, но и по случайным причинам. Но даже в таком случае упадёт следующий (мутирующий) вызов remove. Тем не менее предлагаю написать комментарий и оставить тудушку убрать || через некоторое время.

Comment (l0kix2) [reply pkg/components/master.go:162]: я подумал детектить конкретную ошибку будет оверкилл
давай напишу тудушку

File: pkg/components/master.go
Comment (savnadya) [inline pkg/components/master.go:163]: It will not work. Init job runs only during cluster initialization, and it won't run again during update. You need to create one more job, which will be run after `UpdateStatusWaitingForPodsCreation`.

File: pkg/components/master.go
Comment (zlobober) [inline pkg/components/master.go:198]: We have a nice conventions for the compatibility code that must be erased in some future. It must be prepended with a comment of form `COMPAT(username): remove something when something happens`. The important part is to specify explicitly, what condition must hold in order to remove this COMPAT; this allows code maintainers to periodically grep codebase for the word "COMPAT" and easily understand, which COMPATs may be safely removed without trying to remember the details of the code 6 months ago.

In your case I'd write
```
// COMPAT(l0kix2): remove || part when the compatibility with 23.1 and older is dropped.
```
(also, mind the dot in the end of the line)

File: pkg/components/master.go
Comment (zlobober) [inline pkg/components/master.go:285]: Master exited read-only state

**Review by zlobober — APPROVED** [review]
LGTM modulo cosmetics

**Comment by l0kix2** [comment]
I still need to run some tests before merge anyway. I've ran the case when we deploy ytsaurus cluster and then update base image and everything went as expected. But I'm not sure yet about the case when we deploy cluster with vversion A, then update it to version B and then update to version C (that time need to make sure that existing job will be correctly restarted), but for now i've had some problems, because some of init scripts hanged and complains about master unavailability and I've not yet localized is problem with my local setup or ytsaurus image version or the new code.

**Comment by l0kix2** [comment]
- Fixed codestyle notes
- Successively ran e2e tests
- Tested new exit read only job switching from `ytsaurus/ytsaurus:stable-23.1.0` to `ytsaurus/ytsaurus:stable-23.1.0-relwithdebinfo` and back (script just gracefully fails for that version since cmd is not supported, though job workflow looks fine).

I'll try to test against 23.2 if I be able to find stable enough images.

**Comment by l0kix2** [comment]
- I've tested operator case "deploy verA, update to ver B, update to ver C" against both 23.1 and 23.2 ytsaurus versions and it works as expected.
- Also added check for YTsaurus version in `master_exit_read_only` job script, so now we either do nothing for versions <=23.1 and or run `master_exit_read_only` cmd otherwise (instead of suppressing possible errors).

**Pull request closed without merging**
